src/api/artifact/abstractor/abstractor.go

```diff
@@ -40,6 +42,12 @@ type Abstractor interface {
 	// Abstract the specific information for the specific artifact type into the artifact model,
 	// the information can be got from the manifest or other layers referenced by the manifest.
 	Abstract(ctx context.Context, artifact *artifact.Artifact) error
+	// GetAddition returns the addition of the artifact.
+	// The additions are different for different artifacts:
+	// build history for image; values.yaml, readme and dependencies for chart, etc
+	GetAddition(ctx context.Context, artifact *artifact.Artifact, additionType string) (addition *resolver.Addition, err error)
```

Comment:
Why addition related operations are defined in abstractor, not in the artifact(mgr)?

Reply:
The abstractor extracts the additional contents to the artifact and the related getting/listing OPs should be from artifact side.